File: 179-largest-number/largest-number.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {

        vector<string>ans;
        for(auto &it :nums){
            ans.push_back(to_string(it));
        }
        for(int i=0;i<ans.size()-1;i++){
            for(int j=0;j<ans.size()-i-1;j++){
                string first=ans[j]+ans[j+1];
                string second=ans[j+1]+ans[j];
                if(second>first){
                    swap(ans[j],ans[j+1]);
                }
            }
        }
        string ans1;
        for(int i=0;i<ans.size();i++){
            ans1+=ans[i];
        }
        if(ans1[0]=='0')return "0";
        return ans1;
    }
};

File: 179-largest-number/largest-number.java
class Solution {
 
    public String largestNumber(int[] nums) {
      List<String> arr=new ArrayList<>();
        for(int it :nums){
            arr.add(Integer.toString(it));
            
        }
      Collections.sort(arr,(a,b)->{
        return (b+a).compareTo(a+b);
      });
        String ans1="";
        for(int i=0;i<arr.size();i++){
            ans1+=arr.get(i);
        }
        if(ans1.charAt(0)=='0')return "0";
        return ans1;
     

    }
}